doc/source/contributing_docstring.rst

```diff
@@ -0,0 +1,675 @@
+.. _docstring:
+
+======================
+pandas docstring guide
+======================
+
+About docstrings and standards
+------------------------------
+
+A Python docstring is a string used to document a Python function or method,
+so programmers can understand what it does without having to read the details
+of the implementation.
+
+Also, it is a commonn practice to generate online (html) documentation
+automatically from docstrings. `Sphinx <http://www.sphinx-doc.org>`_ serves
+this purpose.
+
+Next example gives an idea on how a docstring looks like:
+
+.. code-block:: python
+
+    def add(num1, num2):
+    """Add up two integer numbers.
+
+    This function simply wraps the `+` operator, and does not
+    do anything interesting, except for illustrating what is
+    the docstring of a very simple function.
+
+    Parameters
+    ----------
+    num1 : int
+        First number to add
+    num2 : int
+        Second number to add
+
+    Returns
+    -------
+    int
+        The sum of `num1` and `num2`
+
+    See Also
+    --------
+    subtract : Subtract one integer from another
+
+    Examples
+    --------
+    >>> add(2, 2)
+    4
+    >>> add(25, 0)
+    25
+    >>> add(10, -10)
+    0
+    """
+    return num1 + num2
+
+Some standards exist about docstrings, so they are easier to read, and they can
+be exported to other formats such as html or pdf.
+
+The first conventions every Python docstring should follow are defined in
+`PEP-257 <https://www.python.org/dev/peps/pep-0257/>`_.
+
+As PEP-257 is quite open, and some other standards exist on top of it. In the
+case of pandas, the numpy docstring convention is followed. The conventions is
+explained in this document:
+
+- `numpydoc docstring guide <http://numpydoc.readthedocs.io/en/latest/format.html>`_
+  (which is based in the original `Guide to NumPy/SciPy documentation
+  <https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt>`_)
+
+numpydoc is a Sphinx extension to support the numpy docstring convention.
+
+The standard uses reStructuredText (reST). reStructuredText is a markup
+language that allows encoding styles in plain text files. Documentation
+about reStructuredText can be found in:
+
+- `Sphinx reStructuredText primer <http://www.sphinx-doc.org/en/stable/rest.html>`_
+- `Quick reStructuredText reference <http://docutils.sourceforge.net/docs/user/rst/quickref.html>`_
+- `Full reStructuredText specification <http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html>`_
+
+The rest of this document will summarize all the above guides, and will
+provide additional convention specific to the pandas project.
+
+.. _docstring.tutorial:
+
+Writing a docstring
+-------------------
+
+.. _docstring.general:
+
+General rules
+~~~~~~~~~~~~~
+
+Docstrings must be defined with three double-quotes. No blank lines should be
+left before or after the docstring. The text starts immediately after the
```

Comment:
This means the vast majority of current pandas docstring do it wrong (and as a result... are more readable, I think). Everybody all right with this?

Reply:
Yes, I personally also like how we mostly start on the following line (which alsi gives you 3 characters more for the summary line ... :-))

PEP257 also says both are ok.

Reply:
I think I got that from the numpy convention. I'm more used to start in the same line, but as far as we always use the same, I don't think it'll make a difference for anyone.